### ISSUE TITLE
Vega crashes on `pulse.mod.length` in 5.7.1

### ISSUE THREAD

**Issue by anagrius** [vega/vega#2070]
Since 5.7.1 vega crashes on specs that where previously working. It crashes due to code introduced in #2048 

It occurs on line Relay.js#49.

 `pulse.mod.length`

It seems `mod` is not always defined.

![image](https://user-images.githubusercontent.com/157210/66353219-c5bb5980-e961-11e9-92ca-fdb67dd462a9.png)

**Comment by jheer** [comment]
Thank you for sharing the issue. Can you please share a spec that replicates the issue? That way I can check not just this specific line, but any potential upstream bugs that might propagate forward, and also add appropriate test cases as needed.

**Comment by anagrius** [comment]
I wish I could. We have some restrictions preventing us from sharing the original. I will try to get an SSCCE your way. I would use the Vega editor, but it has not been updated yet :)

**Comment by jheer** [comment]
Thanks! I'm curious to see how an "empty" pulse (most likely the StopPropagation object?) made its way into the relay...

**Comment by anagrius** [comment]
@jheer 

Here is a minimal example that reproduces the behaviour:

```
              "data": [
                    {
                      "name": "points"
                    },
                    {
                      "name": "derived1",
                      "source": "points"
                    },
                    {
                      "name": "derived2",
                      "source": "points"
                    },
                    {
                      "name": "joined",
                      "source": [
                        "derived1",
                        "derived2"
                      ],
                      "transform": [
                        {
                          "type": "formula",
                          "expr": "123",
                          "as": "xyz"
                        }
                      ]
                    }
               ]
```

Our real code filters and transformations the values in `derived1` and `derived2`. We use it to "bundle" all the low series into an "Others" series while letting the high valued series pass through in the other, filtering out all the low series. Then joining the streams to get, e.g. `[Series1, Series2, Others]`

**Comment by anagrius** [comment]
If there is anything more I can do to help, please let me know. The bug is pretty bad for us since we don't have a record of the exact vega packages that did work ( 🤦‍♂). Since the upgrade we have not been able to downgrade to a working setup.

**Comment by jheer** [comment]
Thanks! The issue was due to improper multi-pulse handling, when the Relay transform also performs a union of multiple input data sources.

If you revert to 5.7.0 this problem should go away. I've also written a fix that closes this issue, and plan to release a patched 5.7.2 within the next day.
